### ISSUE TITLE
.NET 7.0 RC 2

### ISSUE THREAD

**Issue by dcwhittaker** [dotnet/core#7865]
[Release Notes](https://github.com/dotnet/core/blob/main/release-notes/7.0/preview/7.0.0-rc.2.md)

Please report any issues you find with [.NET 7 RC2](https://devblogs.microsoft.com/dotnet/announcing-dotnet-7-rc-2/), either responding to this issue, creating a new issue or creating a new issue in one of the following repos:

* [dotnet/aspnetcore](https://github.com/dotnet/aspnetcore/issues) - for ASP.NET Core questions
* [dotnet/efcore](https://github.com/dotnet/efcore/issues) - for Entity Framework Core questions
* [dotnet/sdk](https://github.com/dotnet/sdk/issues) - for CLI tools and questions
* [dotnet/runtime](https://github.com/dotnet/runtime/issues) - for runtime, API, or installer issues and questions
* [dotnet/winforms](https://github.com/dotnet/winforms/issues) - for WinForms issues
* [dotnet/wpf](https://github.com/dotnet/wpf/issues) - for WPF issues
* [nuget/home](https://github.com/nuget/home/issues) - for NuGet questions and issues

### Known Issues
If there are any issues with this release we will track them here and check issues off as they're resolved. See the linked issues for details on progress and resolution details.

**Comment by gdar91** [comment]
F# / C# interop fails when upgrading to the RC2.
To reproduce, create a solution with two – F# Library and C# Console projects. Reference the F# project from the C# one. Add at least two types in the Library.fs file (two single case discriminated unions for simplicity…) and access THE FIRST ONE from the C# project. It will throw: ” Program.cs(1, 24): [CS0117] ‘Test.A’ does not contain a definition for ‘A'”

**Comment by gdar91** [comment]
> F# / C# interop fails when upgrading to the RC2. To reproduce, create a solution with two – F# Library and C# Console projects. Reference the F# project from the C# one. Add at least two types in the Library.fs file (two single case discriminated unions for simplicity…) and access THE FIRST ONE from the C# project. It will throw: ” Program.cs(1, 24): [CS0117] ‘Test.A’ does not contain a definition for ‘A'”

[https://github.com/gdar91/Rc2FsCs](https://github.com/gdar91/Rc2FsCs)

**Comment by gortok** [comment]
For [.NET 7 RC 2 release you mention CVE-2022-38013](https://github.com/dotnet/core/blob/main/release-notes/7.0/preview/7.0.0-rc.2.md), but for [.NET 6 and .NET Core 3.1 release (3.1.30 and 6.0.10)](https://github.com/dotnet/core/blob/main/release-notes/3.1/3.1.30/3.1.30.md), you keep the exact same release text but reference an newer CVE, CVE-2022-41032.

Should the .NET 7 RC2 release notes also reference CVE '41032?

**Comment by dcwhittaker** [comment]
@gortok yes you are absolutely right - making those changes now. Thanks for the flag!

**Comment by leecow** [comment]
@KevinRansom or @jaredpar - do you know who should have a look at the F# to C# interop issue reported above (https://github.com/dotnet/core/issues/7865#issuecomment-1275325404)?

**Comment by mirespace** [comment]
Hi team,

(ofyi) I built dotnet7 RC2  amd64 today on Ubuntu (tarball creation included), and I got a different number for the version : 
```
ubuntu@dotnet-jammy:~$ dotnet --info
.NET SDK:
 Version:   7.0.100-rc.2.22517.1
 Commit:    0a5360315a

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         ubuntu.22.04-x64
 Base Path:   /usr/lib/dotnet/dotnet7-7.0.100-rc.2.22477.23/sdk/7.0.100-rc.2.22517.1/
```

I checked into the debian folder to see if that number was harcoded (It's strange to me that it could happen, but just double-checking), and it is not.

**Comment by jaredpar** [comment]
I can repro @gdar91 issue. 

- Succeeded with RC1 SDK
- Failed with RC2 SDK

I took a quick look at the F# binary produced by the RC2 SDK and it seems fine. At the same time I don't think we changed anything in C# import that would intersect with this. 

@cston PTAL.

**Comment by cston** [comment]
Regarding the issue reported by @gdar91 in https://github.com/dotnet/core/issues/7865#issuecomment-1275325404:

For the F# type `type A = A` previously the F# compiler emitted a type with a static property `A` and a `get_A()` method. With RC2 SDK, the type includes a `get_A()` method but no property.

**Comment by KevinRansom** [comment]
Opened an issue in F#:  https://github.com/dotnet/fsharp/issues/14147

**Comment by vzarytovskii** [comment]
Reported earlier: https://github.com/dotnet/fsharp/issues/14088
Fix has been backported to 17.4
